File: 334-increasing-triplet-subsequence/334-increasing-triplet-subsequence.java
class Solution {
    
  public boolean increasingTriplet(int[] nums) {
        int num1 = Integer.MAX_VALUE, num2 = Integer.MAX_VALUE;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] > num2) return true;
            else if(nums[i] > num1) num2 = nums[i];
            else num1 = nums[i];
        }
        return false;
    }
}

File: 334-increasing-triplet-subsequence/334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        a,b = float('inf') , float('inf') 
        
        for num in nums:
            
            if num  > b:
                return True 
            if num < a:
                a = num 
            if num > a:
                b = min(b,num)
            a = min(a,num)
        
        return False 